### PR TITLE
use single chan to emit events and errors, support new struct inode *time fields

### PR DESCRIPTION
**Pull request by mmat11**
- Use single chan to emit events and errors. Context: https://github.com/elastic/beats/pull/37640/files#r1456682398
- Support new struct inode *time fields. Context: https://github.com/elastic/ebpf/pull/182

### CODE REVIEW

File: loader.go [hunk 1]
```diff
@@ -261,43 +261,44 @@ func (l *Loader) attachBpfProgs() error {
...
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case record := <-in:
-			event, err := NewEvent(record.RawSample)
+		default:
+			var r Record
+
+			record, err := l.reader.Read()
```
Comment (pkoutsovasilis) [inline loader.go:277]: here `Read` ends up utilising an epoll wait with a timeout -1 if the reader deadline is not set, which I think we do not set. So just checking here, we are okay ctx.Done being essentially ignored when we get to Read as this is blocking until either an event is produced or something closing the reader?

Comment (mmat11) [reply loader.go:277]: yeah in theory it's fine, a running system will always generate events. I'm open to alternatives though

Comment (mmat11) [reply loader.go:281]: pushed some changes, let me know if it looks good

**Review by mjwolf — APPROVED** [review]

**Review by pkoutsovasilis — COMMENTED** [review]
LGTM

**Pull request merged**